### PR TITLE
fix(http): Fix headers doc to consider major protocol versions

### LINKED ISSUE #20744: HTTP headers landing page makes overly specific references to HTTP/1.1
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers

### What specific section or headline is this issue about?

The intro section

### What information was incorrect, unhelpful, or incomplete?

> HTTP headers let the client and the server pass additional information with an HTTP request or response. An HTTP header consists of its case-insensitive name followed by a colon (:), then by its value. [Whitespace](https://developer.mozilla.org/en-US/docs/Glossary/Whitespace) before the value is ignored.

This is overly specific to how HTTP/1.1 formats fields. 

In RFC 9110 (HTTP semantics), the definition of HTTP fields has been made more general and then specific serialization concerns have been delegated to each HTTP version document (HTTP/1.1 uses one format, HTTP/2 and HTTP/3 use a different one).

### What did you expect to see?

We probably want to introduce headers more generally - before then talking about the specifics of how headers are serialized in each HTTP version. This needs some care to ensure it is logically consistent, and that the references to RFCs are directly related to the subject matter.

### Do you have any supporting links, references, or citations?

https://httpwg.org is the most canonical source

### Do you have anything more you want to share?

Given that this probably needs to incorporate mention of HTTP/2 and HTTP/3, it might make sense to do it as part of that project.

One thing that *will* stick out to developers is that implmentations (browsers, wireshark, curl, servers, etc ) tend to pick a presentation format for HTTP header fields that looks like HTTP/1.1. However, "on the wire" the fields are sent in a potentially different format. By fixing up this page, we'll stumble over that problem and need to take care to present things clearly to the reader.

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/http/headers`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/http/headers/index.md
* Last commit: https://github.com/mdn/content/commit/1fb2843562076484f518705129ba9b64cabad30e
* Document last modified: 2022-09-12T20:36:22.000Z

</details>

### PR DESCRIPTION
**Pull request by bsmth**
### Description

The HTTP headers page can be more consistent across major protocol versions. This is a small PR to tidy up the language, seeing as more detailed improvements went into the messages page recently.

### Related issues and pull requests

Fixes #20744

### CODE REVIEW

File: files/en-us/web/http/headers/index.md [hunk 1]
```diff
@@ -7,10 +7,13 @@ page-type: landing-page
 
 {{HTTPSidebar}}
 
-**HTTP headers** let the client and the server pass additional information with an HTTP request or response. An HTTP header consists of its case-insensitive name followed by a colon (`:`), then by its value. {{Glossary("Whitespace")}} before the value is ignored.
```
Comment (hamishwillee) [inline files/en-us/web/http/headers/index.md:10]: The information about Whitespace being ignored is dropped. Is this OK/intentional?

File: files/en-us/web/http/headers/index.md [hunk 2]
```diff
@@ -7,10 +7,13 @@ page-type: landing-page
...
+**HTTP headers** let the client and the server pass additional information with a message in a request or response.
+In HTTP/1.X, a header is a case-insensitive name followed by a colon, then by its value like `Allow: POST`.
```
Comment (hamishwillee) [inline files/en-us/web/http/headers/index.md:11]: For some reason the "like" here feels wrong. I also thought it worth injecting that optional whitespace. To concider
```suggestion
In HTTP/1.X, a header is a case-insensitive name followed by a colon, then optional whitespace which will be ignored, and finally by its value (for example: `Allow: POST`).
```

**Review by hamishwillee — APPROVED** [review]
@bsmth A couple of comments for you to review. Approving so you are not blocked.

File: files/en-us/web/http/headers/index.md (hunk 1 above)
Comment (hamishwillee) [reply files/en-us/web/http/headers/index.md:10]: FYI added below in https://github.com/mdn/content/pull/37194#discussion_r1883049823

File: files/en-us/web/http/headers/index.md (hunk 2 above)
Comment (bsmth) [reply files/en-us/web/http/headers/index.md:11]: Much better, tnx

File: files/en-us/web/http/headers/index.md (hunk 1 above)
Comment (bsmth) [reply files/en-us/web/http/headers/index.md:10]: Thank you, I'll apply a similar "like" fix for the next sentence

File: files/en-us/web/http/headers/index.md [hunk 3]
```diff
@@ -7,10 +7,13 @@ page-type: landing-page
 
 {{HTTPSidebar}}
 
-**HTTP headers** let the client and the server pass additional information with an HTTP request or response. An HTTP header consists of its case-insensitive name followed by a colon (`:`), then by its value. {{Glossary("Whitespace")}} before the value is ignored.
+**HTTP headers** let the client and the server pass additional information with a message in a request or response.
+In HTTP/1.X, a header is a case-insensitive name followed by a colon, then optional whitespace which will be ignored, and finally by its value (for example: `Allow: POST`).
+In HTTP/2 and above, headers are displayed in lowercase when viewed in developer tools, like `accept: */*`, and prefixed with a colon for a special group of [pseudo-headers](/en-US/docs/Web/HTTP/Messages#pseudo-headers), like `:status: 200`.
```
Comment (bsmth) [inline files/en-us/web/http/headers/index.md:12]:
```suggestion
In HTTP/2 and above, headers are displayed in lowercase when viewed in developer tools (`accept: */*`), and prefixed with a colon for a special group of [pseudo-headers](/en-US/docs/Web/HTTP/Messages#pseudo-headers) (`:status: 200`).
```

**Comment by bsmth** [comment]
Thanks a lot Hamish 👍🏻

**Pull request merged**